### DISCUSSION TITLE
怎么获取一整个文件夹的直链啊

### DISCUSSION THREAD

**Discussion by gqyanann** [spencerwooo/onedrive-vercel-index#510] [Q&A] [answered]
如题，想要不下载下来只要那个直链

**Comment by gqyanann** [comment, marked as answer]
> 如题，想要不下载下来只要那个直链
准确说是那个生成的压缩包的

**Reply by zkeq** [reply]
你看一下控制台，应该可以发现点击下载文件夹后，出现了很多请求

也就是说你点击 `下载按钮` 之后，首先执行的操作是把这些文件全部下载到浏览器的缓存中

全部下载完成之后再进行压缩操作的....也就不存在生成压缩包直链这一说...

**Reply by gqyanann** [reply]
也就是说，如果我想远程下载一整个文件夹的文件的话，只能一个一个文件单独直链的添加吗

**Reply by zkeq** [reply]
https://user-images.githubusercontent.com/62864752/155873975-00b9e193-0ff5-4b04-9496-941dcf0c71c9.mp4

这个我不大清楚...昨天我也想实现这功能，用 `Python` 写了半天

还是放弃了...小文件太多了

把 3000 个文件创建 3000 个 IDM 下载任务太不现实了...

(自己写下载进度条感觉也很麻烦而且这 `API` 好想还有些费用支出...)

**Reply by gqyanann** [reply]
看来我还是老老实实下本地再上传吧

**Reply by myl7** [reply]
这个文件夹下载是 client-side 实现的，因为整个下载过程不能走 Vercel（不然费用支出太大），而直接从 OneDrive 下载的话 OneDrive 没有提供文件夹下载 API 的（实际是存在的，可以借由 OneDrive share link 得知，但似乎是使用的额外单独的服务器，且不便于逆向 API），因此只能在 client-side 合并了。
